web-external/src/components/fileSelector/index.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import content from './content.jade';
```

Comment:
The basic approach is fine, but I'd like to see this better integrated with the rest of the React app.

Instead of appending from a jade (pug?) template to the dom, and using a global selector in the component, the component can use an anonymous div element that serves as the root of the FileSelectorWidget and use a [ref](https://facebook.github.io/react/docs/refs-and-the-dom.html) to access it internally.  Thereby completely wrapping the widget as a component.